src/Databases/DDLLoadingDependencyVisitor.cpp

```diff
@@ -21,11 +21,11 @@ namespace DB
 
 using TableLoadingDependenciesVisitor = DDLLoadingDependencyVisitor::Visitor;
 
-TableNamesSet getLoadingDependenciesFromCreateQuery(ContextPtr global_context, const QualifiedTableName & table, const ASTPtr & ast)
+TableNamesSet getLoadingDependenciesFromCreateQuery(ContextPtr global_context, const QualifiedTableName & table, const ASTPtr & ast, const String & default_database)
 {
     assert(global_context == global_context->getGlobalContext());
     TableLoadingDependenciesVisitor::Data data;
-    data.default_database = global_context->getCurrentDatabase();
+    data.default_database = default_database;
```

Comment:
Previously here we used current database from global context and it was `default`, and it led to wrong dependencies when database is not specified in the query for dependencies

Reply:
We should always use global context in places like this, because there's no local context on server startup. We had many bugs like this:
 - a table is created using a local (session/query) context with modified settings or current database
 - the table works
 - server restarts
 - it doesn't remember the local context that was used for creating the table
 - the table works differently or doesn't work